### PR TITLE
updating deps

### PR DESCRIPTION
**Pull request by beardedeagle**
- Updated all deps, includuding ecto to the latest 2.1 codebase. (2.1.6)
- removed editor section from gitignore
- added more recent elixir/otp version to travis config
- removed credo/dogma and all references to it
- No longer counting irrelevant lines in coverage reports (didn't change a lot, but nice to have)
- logger debug calls are now lazy
- fixed all compile time warnings
- fixed all unit tests
- fixed a couple of spelling errors in the README

### CODE REVIEW

File: test/unit/match_spec_test.exs [hunk 1]
```diff
@@ -193,7 +196,9 @@ defmodule EctoMnesia.Context.MatchSpecTest do
   describe "having" do
     test "is not supported" do
       assert_raise Ecto.Query.CompileError, "`Ecto.Query.having/3` is not supported by Mnesia adapter.", fn ->
-        ms(from(p in "sell_offer") |> having(status: "foo"))
+        sell_offer = from(p in "sell_offer")
+
+        ms(sell_offer |> having(status: "foo"))
```
Comment (AndrewDryga) [inline test/unit/match_spec_test.exs:201]: Use a function call when a pipeline is only one function long

File: test/unit/match_spec_test.exs [hunk 2]
```diff
@@ -84,7 +85,9 @@ defmodule EctoMnesia.Context.MatchSpecTest do
 
   describe "where" do
     test "in expression style" do
-      query = from(so in "sell_offer") |> where(status: "foo")
+      sell_offer = from(so in "sell_offer")
+
+      query = sell_offer |> where(status: "foo")
```
Comment (AndrewDryga) [inline test/unit/match_spec_test.exs:90]: Use a function call when a pipeline is only one function long

File: test/unit/match_spec_test.exs [hunk 3]
```diff
@@ -27,12 +27,13 @@ defmodule EctoMnesia.Context.MatchSpecTest do
...
       age = 25
+      sell_offer = from(so in "sell_offer")
 
-      query = from(so in "sell_offer") |> where(status: ^status)
+      query = sell_offer |> where(status: ^status)
       assert [{_, [{:==, @status_field_id, ^status}], _}] = ms(query)
 
       # Multiple sources
-      query = from(so in "sell_offer") |> where(status: ^status, age: ^age)
+      query = sell_offer |> where(status: ^status, age: ^age)
```
Comment (AndrewDryga) [inline test/unit/match_spec_test.exs:36]: Use a function call when a pipeline is only one function long

File: test/unit/match_spec_test.exs [hunk 4]
```diff
@@ -27,12 +27,13 @@ defmodule EctoMnesia.Context.MatchSpecTest do
     test "in expression style" do
       status = "foo"
       age = 25
+      sell_offer = from(so in "sell_offer")
 
-      query = from(so in "sell_offer") |> where(status: ^status)
+      query = sell_offer |> where(status: ^status)
```
Comment (AndrewDryga) [inline test/unit/match_spec_test.exs:32]: Use a function call when a pipeline is only one function long

**Comment by AndrewDryga** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 17 possible new issues (including those that may have been commented here).
* 9 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/Nebo15/ecto_mnesia/pulls/59.

**Comment by beardedeagle** [comment]
Splitting #58 into 2 separate PR's. This is the first, which brings ecto_mnesia current where we can go latest. Once this is in I will put in a separate PR for the formatter run and close out the original PR.

**Review by AndrewDryga — APPROVED** [review]

**Comment by AndrewDryga** [comment]
Awesome job @beardedeagle

**Pull request merged**